topography/include/gaussian_topog.inc

```diff
@@ -162,112 +59,53 @@ end subroutine gaussian_topog_init
 !! <br>Example usage:
 !! @code{.F90} zsurf = <B>get_gaussian_topog</B> ( lon, lat, height
 !!                    [, olond, olatd, wlond, wlatd, rlond, rlatd ] )@endcode
-function get_gaussian_topog ( lon, lat, height,                          &
-                              olond, olatd, wlond, wlatd, rlond, rlatd ) &
-                     result ( zsurf )
 
-real, intent(in)  :: lon(:), lat(:)
-real, intent(in)  :: height
-real, intent(in), optional :: olond, olatd, wlond, wlatd, rlond, rlatd
-real :: zsurf(size(lon,1),size(lat,1))
+function GET_GAUSSIAN_TOPOG_(lon, lat, height,                         &
+                            olond, olatd, wlond, wlatd, rlond, rlatd ) &
+                     result (zsurf )
+
+real(kind=FMS_TOP_KIND_), intent(in)  :: lon(:), lat(:)
+real(kind=FMS_TOP_KIND_), intent(in)  :: height
+real(kind=FMS_TOP_KIND_), intent(in), optional :: olond, olatd, wlond, wlatd, rlond, rlatd
+real(kind=FMS_TOP_KIND_) :: zsurf(size(lon,1),size(lat,1))
 
 integer :: i, j
-real    :: olon, olat, wlon, wlat, rlon, rlat
-real    :: tpi, dtr, dx, dy, xx, yy
+real(kind=FMS_TOP_KIND_)    :: olon, olat, wlon, wlat, rlon, rlat
+real(kind=FMS_TOP_KIND_)    :: tpi, dtr, dx, dy, xx, yy
+integer, parameter     :: lkind = FMS_TOP_KIND_
 
   if (do_nml) call read_namelist
 
 ! no need to compute mountain if height=0
-  if ( height == 0. ) then
-       zsurf(:,:) = 0.
+  if ( height == 0.0_lkind) then
+       zsurf(:,:) = 0.0_lkind
        return
   endif
 
-  tpi = 2.0*pi
-  dtr = tpi/360.
+  tpi = 2.0_lkind*real(pi, FMS_TOP_KIND_)
+  dtr = tpi/360.0_lkind
 
 ! defaults and convert degrees to radians (dtr)
-  olon = 90.*dtr;  if (present(olond)) olon=olond*dtr
-  olat = 45.*dtr;  if (present(olatd)) olat=olatd*dtr
-  wlon = 15.*dtr;  if (present(wlond)) wlon=wlond*dtr
-  wlat = 15.*dtr;  if (present(wlatd)) wlat=wlatd*dtr
-  rlon =  0.    ;  if (present(rlond)) rlon=rlond*dtr
-  rlat =  0.    ;  if (present(rlatd)) rlat=rlatd*dtr
+  olon = 90.0_lkind*dtr;  if (present(olond)) olon=olond*dtr
+  olat = 45.0_lkind*dtr;  if (present(olatd)) olat=olatd*dtr
+  wlon = 15.0_lkind*dtr;  if (present(wlond)) wlon=wlond*dtr
+  wlat = 15.0_lkind*dtr;  if (present(wlatd)) wlat=wlatd*dtr
+  rlon = 0.0_lkind     ;  if (present(rlond)) rlon=rlond*dtr
+  rlat = 0.0_lkind     ;  if (present(rlatd)) rlat=rlatd*dtr
 
 ! compute gaussian-shaped mountain
     do j=1,size(lat(:))
       dy = abs(lat(j) - olat)   ! dist from y origin
-      yy = max(0., dy-rlat)/wlat
+      yy = max(0.0_lkind, dy-rlat)/wlat
```

Comment:
all module levels variables are declared as r8_kind.  you need to convert them to lkind